src/scala/microsoft-spark-2.3.x/src/main/scala/org/apache/spark/deploy/dotnet/DotnetRunner.scala

```diff
@@ -232,6 +235,30 @@ object DotnetRunner extends Logging {
     dotnetBackend.close()
   }
 
+  private def closeDotnetProcess(dotnetProcess: Process): Int = {
+    if (dotnetProcess == null) {
+      return -1
+    } else if (!dotnetProcess.isAlive) {
+      return dotnetProcess.exitValue()
+    }
+
+    // Try to (gracefully on Linux) kill the process and resort to force if interrupted
+    var returnCode = -1
+    logInfo("Closing .NET process")
+    try {
+      dotnetProcess.destroy()
```

Comment:
Do we want to call `destroy` before `waitFor` even for the normal exit case? What will happen without this?

Reply:
ah nevermind, this is the second attempt.